### PR TITLE
Update Form's documentation

### LINKED ISSUE #6667: Form's onChangeCallback is not documented.
**Describe the bug**
I was trying to use the `Form` component and do some action whenever anything changes in its state. Coming from the Web Components world, I felt it natural to have change event bubbled/composed. So, at first, I tried to use `onChange` prop which I've seen in some other components. It occurred there is no such thing. Then, I checked the docs https://woocommerce.github.io/woocommerce-admin/#/components/packages/form/README?id=form but there is nothing about change event/callback. Then I tried to extend the component to add this feature. While reading the code to find a way to plug in, I noticed undocumented `onChangeCallback`.

I see few issues here:
1. The `onChangeCallback` is not documented, making it hard to discover and use.
2. Names are inconsistent across components: `onChange` vs. `onChangeCallback`
3. There is a mysterious `touched` prop without any docs.

 
**To Reproduce**

1. Go to https://woocommerce.github.io/woocommerce-admin/#/components/packages/form/README?id=form
2. Try to figure out, how to do something when any value changes.

**Expected behavior**
`onChangeCallback` should be mentioned there, ideally with some meaningful description, examples and documented arguments.

**Screenshots**
Screenshot of current props documentation

**Additional context**
I tried to add it by myself but wasn't able to generate that https://github.com/woocommerce/woocommerce-admin/issues/6098.

### PR DESCRIPTION
**Pull request by hsingyuc**
Fixes #6667 

Update Form's props `touched` and `onChange` documentation. Rename onChangeCallback to onChange

### Detailed test instructions:

-  Check component `Form` README.md

### CODE REVIEW

**Comment by moon0326** [comment]
@hsingyuc Thank you for putting this together 👍 The change looks simple and clean. 

Since `Form` is published package and there might be other teams who are already using `onChangeCallback`, I'm afraid this change might break their code. cc ing @joshuatf for suggestions.

**Comment by tomalec** [comment]
What do you think of making `onChangeCallback` a deprecated alias for `onChange`?

**Comment by samueljseay** [comment]
I apologise for introducing `onChangeCallback` and not documenting it 😓 I needed the feature at the time because `Form` was lacking. I agree we should deprecate it but have it call onChange.

**Comment by joshuatf** [comment]
Should we also deprecate `onSubmitCallback` in favor of `onSubmit`?  @hsingyuc could you take a look at this if you get a chance?

File: packages/components/src/form/index.js [hunk 1]
```diff
@@ -148,8 +161,14 @@ Form.propTypes = {
...
 	 * Function to call when a value changes in the form.
+	 *
+	 * @deprecated
 	 */
 	onChangeCallback: PropTypes.func,
+	/**
+	 * Function to call when a value changes in the form.
+	 */
+	onChange: PropTypes.func,
 	/**
```
Comment (becdetat) [inline packages/components/src/form/index.js:172]: Can you add `touched` as a prop type here?

**Comment by hsingyuc** [comment]
I made the changes!

File: packages/components/src/form/index.js [hunk 2]
```diff
@@ -84,12 +95,23 @@ class Form extends Component {
...
 	async handleSubmit() {
 		const { values } = this.state;
+		const { onSubmitCallback, onSubmit } = this.props;
 		const touched = {};
 		Object.keys( values ).map( ( name ) => ( touched[ name ] = true ) );
 		this.setState( { touched } );
 
 		if ( await this.isValidForm() ) {
-			this.props.onSubmitCallback( values );
+			const callback = onSubmitCallback ? onSubmitCallback : onSubmit;
```
Comment (becdetat) [inline packages/components/src/form/index.js:104]: This should also default to `onSubmit` as the `Callback` prop is deprecated

Comment (hsingyuc) [reply packages/components/src/form/index.js]: sorry, my mistake! The change has been pushed.

**Comment by becdetat** [comment]
Thanks @hsingyuc :-) your changes look good. I'll resolve the conflicts and merge this today.

**Review by becdetat — APPROVED** [review]

**Pull request merged**